### PR TITLE
install.sh: Add brew integration and documentation

### PR DESCRIPTION
**Pull request by nhooyr**
Updates #80

### CODE REVIEW

**Comment by nhooyr** [comment]
I updated the log output with a colored big header for next steps (re https://github.com/terrastruct/d2/pull/114/files)

Please test @alixander 

Screenshot 2022-11-20 at 6 51 20 PM
Screenshot 2022-11-20 at 6 51 27 PM

File: ci/release/_install.sh [hunk 1]
```diff
@@ -149,49 +155,75 @@ main() {
...
+        TALA_VERSION="$( RELEASE_INFO= install_tala_standalone && echo "$VERSION" )"
+      fi
+      ;;
+    homebrew)
+      install_d2_brew
+      if [ -n "${TALA-}" ]; then install_tala_brew; fi
+      ;;
+  esac
+
+  FGCOLOR=2 bigheader 'next steps'
```
Comment (alixander) [inline ci/release/_install.sh:204]: Screen Shot 2022-11-20 at 7 03 46 PM

we should only show this when there are next steps, e.g. d2 not found in path.

File: ci/release/_install.sh [hunk 2]
```diff
@@ -204,36 +236,61 @@ EOF
...
+      log "Run man d2plugin-tala for detailed TALA docs."
     fi
   fi
   logcat >&2 <<EOF
 
 Something not working? Please let us know:
-https://github.com/terrastruct/d2/issues/new
+https://github.com/terrastruct/d2/issues
+https://github.com/terrastruct/d2/discussions
+https://discord.gg/NF6X8K4eDq
```
Comment (alixander) [inline ci/release/_install.sh:249]: kind of overwhelming. just issues i'd say

File: ci/release/_install.sh [hunk 3]
```diff
@@ -252,19 +309,39 @@ install_standalone_d2() {
   "$sh_c" sh -c "'cd \"$INSTALL_DIR/d2-$VERSION\" && make install PREFIX=\"$PREFIX\"'"
 }
 
-install_tala() {
-  REPO="${REPO_TALA:-terrastruct/TALA}"
+install_d2_brew() {
+  header "installing d2 with homebrew"
+  sh_c brew tap terrastruct/d2
+  sh_c brew install d2
+  sh_c brew test d2
```
Comment (alixander) [inline ci/release/_install.sh:316]: my brew gave a warning? r we supposed to run this post-install for everyone?

Screen Shot 2022-11-20 at 7 05 51 PM

**Review by alixander — CHANGES_REQUESTED** [review]
should we turn off brew's autoupdate? https://apple.stackexchange.com/questions/293250/how-to-install-an-homebrew-formula-without-updating-homebrew-itself

my homebrew just started updating random shit, as it does.

File: README.md [hunk 4]
```diff
@@ -92,7 +95,7 @@ curl -fsSL https://d2lang.com/install.sh | sh -s -- --uninstall
 Alternatively, you can install from source:
 
 ```sh
-go install oss.terrastruct.com/d2
```
Comment (alixander) [inline README.md:95]: by default, is this not equivalent to the changed? i.e. does it not get latest by default?

**Review by alixander — COMMENTED** [review]
huh, why did it update my go version. is the dependency set to be the latest go?

Screen Shot 2022-11-20 at 7 11 38 PM

File: install.sh [hunk 5]
```diff
@@ -608,49 +619,75 @@ main() {
   PREFIX=${PREFIX:-/usr/local}
   CACHE_DIR=$(cache_dir)
   mkdir -p "$CACHE_DIR"
+  METHOD=${METHOD:-detect}
   INSTALL_DIR=$PREFIX/lib/d2
 
+  case $METHOD in
+    detect)
+      case "$OS" in
+        macos)
```
Comment (alixander) [inline install.sh:628]: is OS the only criteria for triggering homebrew install? is there a check that `brew` is in their path?

File: README.md (hunk 4 above)
Comment (nhooyr) [reply README.md:95]: The purpose of `@latest` is to force ignoring the current go module. See docs at https://go.dev/doc/go-get-install-deprecation

The other change is using the import path of the d2 executible and not the library.

**Comment by nhooyr** [comment]
> huh, why did it update my go version. is the dependency set to be the latest go?

Fixed, see https://github.com/terrastruct/homebrew-d2/commit/446e29492d3810fb951d123ed3acf3a2f41f3295

File: ci/release/_install.sh (hunk 3 above)
Comment (nhooyr) [reply ci/release/_install.sh:316]: Oh I must have dev mode on by default. Ok, in that case I'll just remove `brew test`.

File: docs/INSTALL.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+brew install d2
+brew test d2
+```
+
+For closed source TALA [https://github.com/terrastruct/tala]:
+
+```sh
+brew tap terrastruct/d2
+brew install tala
+brew test tala
```
Comment (alixander) [inline docs/INSTALL.md:79]: u can probably remove these 2 `brew test` as well. i don't think i've ever seen it in installation instructions for other packages

Comment (nhooyr) [reply docs/INSTALL.md:79]: I never pushed my updates lol

File: ci/release/_install.sh (hunk 1 above)
Comment (nhooyr) [reply ci/release/_install.sh:204]: What part are you referring to? The communication links?

File: ci/release/_install.sh (hunk 2 above)
Comment (nhooyr) [reply ci/release/_install.sh:249]: Can we just remove this please, it's just noise.

Comment (alixander) [reply ci/release/_install.sh:260]: i'm like an 8/10 here that these add value, which is pretty strong but not full insist. if you're higher in your opinion, ok we'll remove.

File: ci/release/_install.sh [hunk 7]
```diff
@@ -492,4 +501,9 @@ fetch_gh() {
   sh_c mv "$file.inprogress" "$file"
 }
 
+# Make's brew sane.
```
Comment (alixander) [inline ci/release/_install.sh:504]: "makes"

**Review by alixander — APPROVED** [review]

File: ci/release/_install.sh (hunk 1 above)
Comment (alixander) [reply ci/release/_install.sh:215]: the "next steps". like there is no next steps. they're done, it's a complete success. the logs after are just "extra"

File: ci/release/_install.sh (hunk 2 above)
Comment (nhooyr) [reply ci/release/_install.sh:260]: Ok, let's leave these but then also always print next steps.

File: ci/release/_install.sh (hunk 1 above)
Comment (nhooyr) [reply ci/release/_install.sh:215]: I disagree, next steps is useful as a summary/intro.

We can debate further later, for now I'll merge and get it in use so we can see how people feel.

**Pull request merged**